portal-kernel/src/com/liferay/portal/kernel/security/permission/propagator/PermissionPropagator.java

```diff
@@ -28,4 +28,9 @@ public void propagateRolePermissions(
 			long[] roleIds)
 		throws PortalException;
 
+	public void propagateRolePermissions(
+			ActionRequest actionRequest, String className, String primKey,
+			String parentClassName, String parentPrimKey, long[] roleIds)
+		throws PortalException;
```

Comment:
It makes me confused. In the existing method, `String className, String primKey` is className and id of the parent